src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -1,58 +1,341 @@
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displaySeriousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getSeriousFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displayDangerousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDangerousFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      showMeQuestion$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecks),
+        select(getShowMeQuestion),
+      ),
+      showMeQuestionOptions$: currentTest$.pipe(
+        select(getTestOutcome),
+        map((outcome) =>
+          getShowMeQuestionOptions(this.showMeQuestions, outcome, this.outcomeBehaviourProvider)),
+      ),
+      tellMeQuestionText$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecks),
+        select(getSelectedTellMeQuestionText),
+      ),
+      etaFaults$: currentTest$.pipe(
+        select(getTestData),
+        select(getETA),
+        select(getETAFaultText),
+      ),
+      ecoFaults$: currentTest$.pipe(
+        select(getTestData),
+        select(getEco),
+        select(getEcoFaultText),
+      ),
+      dangerousFaults$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(category$),
+        map(([testData, category]) =>
+          this.faultSummaryProvider.getDangerousFaultsList(testData, category as TestCategory)),
+      ),
+      seriousFaults$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(category$),
+        map(([testData, category]) =>
+          this.faultSummaryProvider.getSeriousFaultsList(testData, category as TestCategory)),
+      ),
+      drivingFaults$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(category$),
+        map(([testData, category]) =>
+          this.faultSummaryProvider.getDrivingFaultsList(testData, category as TestCategory)),
+      ),
+      drivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        withLatestFrom(category$),
+        map(([testData, category]) => {
+          return this.faultCountProvider.getDrivingFaultSumCount(category as TestCategory, testData);
+        }),
+      ),
```

Comment:
would suggest all of this can live within the base page

Reply:
comments below